PurchasesCoreSwift/SubscriberAttributes/SubscriberAttributesManager.swift

```diff
@@ -0,0 +1,263 @@
+            if syncingAppUserId != currentAppUserId {
+                deviceCache.deleteAttributesIfSynced(appUserID: syncingAppUserId)
+            }
+        } else {
+            let receivedNSError = error as NSError?
+            Logger.error(String(format: Strings.attribution.attributes_sync_error,
+                                receivedNSError?.localizedDescription ?? "",
+                                receivedNSError?.userInfo ?? ""))
+        }
+    }
+
+    @objc public func unsyncedAttributesByKey(appUserID: String) -> SubscriberAttributeDict {
+        // TODO  these logs were in RCPurchases+subscriberattributes, are we ok with logging
+        // them every time?
+        let unsyncedAttributes = deviceCache.unsyncedAttributesByKey(appUserID: appUserID)
+        Logger.debug(String(format: Strings.attribution.unsynced_attributes_count,
+                            unsyncedAttributes.count,
+                            appUserID))
+        if unsyncedAttributes.count > 0 {
+            Logger.debug(String(format: Strings.attribution.unsynced_attributes, unsyncedAttributes))
+        }
+
+        return unsyncedAttributes
+    }
+
+    @objc public func unsyncedAttributesByKeyForAllUsers() -> [String: SubscriberAttributeDict] {
+        return deviceCache.unsyncedAttributesForAllUsers()
+    }
+
+    @objc public func markAttributesAsSynced(_ maybeAttributesToSync: SubscriberAttributeDict?, appUserID: String) {
+        guard let attributesToSync = maybeAttributesToSync,
+              !attributesToSync.isEmpty else {
+            return
+        }
+
+        Logger.info(String(format: Strings.attribution.marking_attributes_synced,
+                           appUserID,
+                           attributesToSync.description))
+
+        lock.lock()
+        var unsyncedAttributes = unsyncedAttributesByKey(appUserID: appUserID)
+        for (key, attribute) in attributesToSync {
+            if let unsyncedAttribute = unsyncedAttributes[key] {
+                if unsyncedAttribute == attribute {
+                    unsyncedAttribute.isSynced = true
+                    unsyncedAttributes[key] = unsyncedAttribute
+                }
+            }
+        }
+        deviceCache.store(subscriberAttributesByKey: unsyncedAttributes, appUserID: appUserID)
+        lock.unlock()
+    }
+
+    @objc public func convertAttributionDataAndSetAsSubscriberAttributes(attributionData: [String: Any],
+                                                                         network: AttributionNetwork,
+                                                                         appUserID: String) {
+        let convertedAttribution = attributionDataMigrator.convertToSubscriberAttributes(
+            attributionData: attributionData,
+            network: network.rawValue)
+        setAttributes(convertedAttribution, appUserID: appUserID)
+    }
+
+    private func storeAttributeLocallyIfNeeded(key: String, value: String?, appUserID: String) {
```

Comment:
style: 
we should create a private extension of the class and move the private methods to it. 
Once there, we can omit the `private` keyword for each `func`

Reply:
this effectively serves to physically separate the public and the private sections

Reply:
ahhhhh yes forgot that one. 👍